### ISSUE TITLE
[BUG][Ascend]  TritonToLinalgIncubated asserts "Cannot destroy a value that still has uses!" (BlockOperand) on valid Triton kernel with nested opaque control flow

### ISSUE THREAD

**Issue by O1iveira7** [flagos-ai/FlagTree#1241]
## Environment
- Triton 3.5.1 (FlagTree 0.7.0rc2+ascend3.5), Ascend (NPU) backend
- torch 2.9.0+cpu + torch_npu 2.9.0.post2, CANN 9.0.0 (driver 25.2.0), Python 3.11.15

## Symptom
触发条件：编译一个合法 Triton kernel，特征为
- 由 _fz_switch_ptr 动态读出的值构造 opaque_true / opaque_false 两个不透明条件；
- opaque_false / opaque_true 交替嵌套（多层 if）；
- 最内层有一个提前 return；
- if 外围有普通的 tl.load / tl.store（mask 读写）。
观察到的行为：
1. 编译进入 ttadapter 阶段（stage 名 ConvertTritonIRToLinalgIR）后，在 TritonToLinalgIncubated pass 上失败。
2. stderr 输出 C++ 断言（稳定、必现）：
.../mlir/IR/UseDefLists.h:198: mlir::IRObjectWithUseList<OperandType>::~IRObjectWithUseList()
  [with OperandType = mlir::BlockOperand]:
  Assertion `use_empty() && "Cannot destroy a value that still has uses!"' failed.
即销毁 Block 时仍存在引用它的 BlockOperand（悬空 use）。
3. 随后报：
note: Pipeline failed while executing [`TritonToLinalgIncubated` on 'builtin.module' operation]
triton.compiler.errors.MLIRCompilationError  (stage ConvertTritonIRToLinalgIR)
4. 没有任何 kernel 在设备上执行，也没有错误输出张量；失败发生在纯编译期。
5. 把最内层的 return 去掉（其余不变）→ 编译通过；把中间的 if opaque_true 去掉 → 编译通过。说明触发点是「嵌套不透明控制流中的提前 return」。

**FlagTree Ascned3.5 output:**
详细请看err_log
```
....
[reproduced] TritonToLinalgIncubated failed while lowering the kernel
             see the assertion above (stderr) and the error below
             MLIRCompilationError:
```

**Triton3.7 RTX3070 output:**
```
[not reproduced] the kernel compiled successfully
output: [0.0, 0.0, 2.5, 0.0, 4.0]
EXIT=1
```

  ## Minimal reproducer
```
import torch
import torch_npu  # noqa: F401  (registers the "npu" device)
import triton
import triton.language as tl

@triton.jit
def relu_kernel(_fz_switch_ptr, _fz_switch_output_ptr, input, output, n_elements, BLOCK_SIZE: tl.constexpr, PADDED_BLOCK_SIZE: tl.constexpr):
    opaque_values = tl.load(_fz_switch_ptr + tl.arange(0, 2))
    opaque_zero, opaque_one = tl.split(opaque_values)
    opaque_true = opaque_zero < opaque_one
    opaque_false = opaque_zero > opaque_one
    pid = tl.program_id(0)
    offset = tl.arange(0, PADDED_BLOCK_SIZE) + pid * BLOCK_SIZE
    if opaque_true:
        x = tl.load(input + offset, (offset < n_elements) & (tl.arange(0, PADDED_BLOCK_SIZE) < BLOCK_SIZE))
    else:
        x = tl.load(input + offset, (offset < n_elements) & (tl.arange(0, PADDED_BLOCK_SIZE) < BLOCK_SIZE))
    if opaque_false:
        _fz_a1_tpl_result = opaque_values.to(tl.float32)
        if opaque_true:
            _fz_a6_sca_v = opaque_values.to(tl.float32)
            _fz_a6_sca_a, _fz_a6_sca_b = tl.split(_fz_a6_sca_v)
            if opaque_false:
                _fz_a8_hs_v = tl.arange(0, 8) & 3
                _fz_a8_hs_m = (_fz_a8_hs_v + opaque_one) % 2 == 0
                _fz_a8_hs_h = tl.histogram((_fz_a8_hs_v + opaque_one) % 4, 4, mask=_fz_a8_hs_m)
                _fz_a8_hs_c = tl.cumsum(_fz_a8_hs_h, axis=0)
                _fz_a8_hs_s = tl.sum(_fz_a8_hs_c)
                _fz_a8_tpl_result = (_fz_a8_hs_s + opaque_values).to(tl.float32)
                tl.store(_fz_switch_output_ptr + 14 + tl.arange(0, 2), _fz_a8_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
                if opaque_false:
                    return
            _fz_a6_sca_x = _fz_a6_sca_a + _fz_a6_sca_b
            _fz_a6_sca_x = _fz_a6_sca_a + _fz_a6_sca_b
            _fz_a6_sca_u = tl.expand_dims(_fz_a6_sca_x, 0)
            _fz_a6_sca_s = tl.reshape(_fz_a6_sca_u, ())
            _fz_a6_sca_b1, _fz_a6_sca_b2 = tl.broadcast(_fz_a6_sca_s, tl.reshape(tl.arange(0, 4), (4, 1)))
            _fz_a6_sca_e = tl.expand_dims(_fz_a6_sca_x, 0)
            _fz_a6_sca_r = tl.broadcast_to(tl.expand_dims(_fz_a6_sca_e, 0), (1, 4))
            _fz_a6_sca_f = tl.sum(_fz_a6_sca_b1 + _fz_a6_sca_r)
            _fz_a6_tpl_result = _fz_a6_sca_f + opaque_values.to(tl.float32)
            tl.store(_fz_switch_output_ptr + 10 + tl.arange(0, 2), _fz_a6_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
        tl.store(_fz_switch_output_ptr + 0 + tl.arange(0, 2), _fz_a1_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
    x = max(0, x)
    tl.store(output + offset, x, (offset < n_elements) & (tl.arange(0, PADDED_BLOCK_SIZE) < BLOCK_SIZE))


def solve(input: torch.Tensor, output: torch.Tensor, N: int) -> None:
    _fz_device = input.device
    _fz_switch_ptr = torch.tensor([0, 1], dtype=torch.int32, device=_fz_device)
    _fz_switch_output_ptr = torch.empty(500, dtype=torch.float32, device=_fz_device)
    BLOCK_SIZE = 1024
    grid = (triton.cdiv(N, BLOCK_SIZE),)
    relu_kernel[grid](
        _fz_switch_ptr,
        _fz_switch_output_ptr,
        input,
        output,
        N,
        BLOCK_SIZE,
        PADDED_BLOCK_SIZE=triton.next_power_of_2(BLOCK_SIZE),
        # debug=True makes MLIR print the failing pass, the assertion and a
        # crash reproducer; the failure itself also happens without it.
        debug=True,
    )
    torch.npu.synchronize()


def main() -> int:
    assert torch.npu.is_available()
    input = torch.tensor(
        [-1.0, 0.0, 2.5, -3.5, 4.0],
        dtype=torch.float32,
        device="npu",
    )
    output = torch.empty_like(input)
    try:
        solve(input, output, 5)
    except Exception as error:
        if "ConvertTritonIRToLinalgIR" in str(error):
            print("[reproduced] TritonToLinalgIncubated failed while lowering the kernel")
            print("             see the assertion above (stderr) and the error below")
            print(f"             {type(error).__name__}: {str(error).splitlines()[0]}")
            return 0
        raise
    print("[not reproduced] the kernel compiled successfully")
    return 1


if __name__ == "__main__":
    raise SystemExit(main())
 ```
[err_log.txt](https://github.com/user-attachments/files/32438425/err_log.txt)
